src/transformers/visitors/toCircuitVisitor.ts

```diff
@@ -164,12 +164,48 @@ const visitor = {
   },
 
   ParameterList: {
-    enter(path: NodePath) {
-      const { node, parent } = path;
+    enter(path: NodePath, state: any) {
+      const { node, parent, scope } = path;
+      let returnName : string;
+       if(path.key === 'parameters'){
       const newNode = buildNode('ParameterList');
       node._newASTPointer = newNode.parameters;
       parent._newASTPointer[path.containerName] = newNode;
-    },
+    } else if(path.key === 'returnParameters'){
+       parent.body.statements.forEach(node => {
+        if(node.nodeType === 'Return'){
+
+          // console.log(referencedId);
+          for(const [ id , bindings ] of Object.entries(scope.referencedBindings)){
+            if(id == node.expression.referencedDeclaration) {
+              if ((bindings instanceof VariableBinding))
+            state.returnIsSecret =bindings.isSecret
```

Comment:
could we make sure only to add the return statement to the circuit if the state `interactsWithSecret`? If you need help let me know :)